src/app/doc/parts/start/start.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, OnInit } from '@angular/core';
+import { environment } from 'src/environments/environment';
+
+@Component({
+  selector: 'app-start',
+  templateUrl: './start.component.html',
+  styleUrls: ['./start.component.scss']
+})
+export class StartComponent implements OnInit {
+
+  public readonly CODES = {
+    ERROR_MSG_1: `npm WARN ngx-smart-modal@x.x.x requires a peer of web-animations-js@>=x.x.x but none was installed.`,
+    ERROR_MSG_2: `warning "ngx-smart-modal@x.x.x" has unmet peer dependency "web-animations-js@>=x.x.x".`,
+    SYSTEM_JS: `System.config({
+  map: {
+    'ngx-smart-modal': 'node_modules/ngx-smart-modal/bundles/ngx-smart-modal.umd.js'
+  }
+});`,
+    MODULE_IMPORT: `import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { NgxSmartModalModule } from 'ngx-smart-modal';
+
+import { AppComponent } from './app.component';
+
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule,
+    NgxSmartModalModule.forRoot()
+  ],
+  providers: [ ],
+  bootstrap: [ AppComponent ]
+})
+export class AppModule { }`,
+    CSS_IMPORT: `/* You can add global styles to this file, and also import other style files */
+@import "~ngx-smart-modal/ngx-smart-modal";
+@import "app/app.component";
+// ...`,
+    MODAL_HTML: `<ngx-smart-modal #myModal identifier="myModal">
+  <h1>Title</h1>
+  <p>Some stuff...</p>
+
+  <button (click)="myModal.close()">Close</button>
+</ngx-smart-modal>`,
+    MODAL_COMP: `import { Component } from '@angular/core';
+import { NgxSmartModalService } from 'ngx-smart-modal';
+
+@Component({
+  // ...
+})
+export class AppComponent {
+  constructor(public ngxSmartModalService: NgxSmartModalService) {
+  }
+}`,
+    INLINE_EXAMPLE: `<button (click)="ngxSmartModalService.getModal('myModal').open()">Open myModal</button>`,
+    MODAL_CLASS_REF: `import { Component } from '@angular/core';
+
+import { MyComponent } from 'xxxx';
+
+import { NgxSmartModalService } from 'ngx-smart-modal';
+
+@Component({
+  // ...
+})
+export class AppComponent {
+  @ViewChild(TemplateRef) tpl: TemplateRef<any>;
```

Comment:
`@ViewChild(TemplateRef, { static: false }) tpl: TemplateRef<any>;`

Reply:
Not important since it's doc and this is asked by IDE during dev. Added as a comment.